src/Forms/Traits/MultipleOptionsTrait.php

```diff
@@ -104,6 +105,42 @@ public function fromQuery(Connection $pdo, $sql, $data = array(), $groupBy = fal
         return $this->fromResults($results, $groupBy);
     }
 
+    /**
+     * Build options array from a DataSet, as provided by <b>Domain</b> gateways
+     * 
+     * @param Dataset $dataset
+     * @param string $valCol
+     * @param string $nameCol
+     * @param string $groupBy
+     * 
+     * @return self
+     */
+    public function fromDataSet(Dataset $dataset, $valCol, $nameCol, $groupBy = false)
+    {
+        if(empty($dataset) || !is_object($dataset)){
+            throw new \InvalidArgumentException(sprintf('Element %s: fromQuery expects value to be an Object, %s given.', $this->getName(), gettype($dataset)));
+        }
+
+        if($dataset->getTotalCount() > 0)
+        {
+            $options = array_filter($dataset->toArray(), function ($item) use ($valCol,$nameCol) {
+                return isset($item[$valCol]) && isset($item[$nameCol]);
+            });
+        }
+
+        foreach ($options as $option) {
```

Comment:
`$options` is being set inside the preceding `if`, so it looks like this loop may also be best inside the same conditional, otherwise the current code could throw a PHP notice for using an unassigned variable.